### PR TITLE
CI Builds with GitHub Actions

### PR DESCRIPTION
**Pull request by ethomson**
I wanted to see if there was any interest in adding GitHub Actions to the party for validating pull request builds and merges into master.

This should be almost the same configuration as what you have for Azure Pipelines and Travis, except with a lack of caching. (Artifact caching is coming 🔜 to GitHub Actions, but I was curious how well it performed even without that.)

/cc @RyanCavanaugh

### CODE REVIEW

**Comment by orta** [comment]
Nice, we've got a few places using actions around our infra:

 - [Website staging/prod](https://github.com/microsoft/TypeScript-Website/blob/master/.github/workflows/build-website.yml)
- [Playground daily builds](https://github.com/orta/make-monaco-builds#monaco-daily-builds)

And an almost ready for publishing blog post on it all 👍

**Comment by RyanCavanaugh** [comment]
Let's see what happens!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      with:
+        fetch-depth: 5
+    - name: Use node version ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: Remove existing TypeScript
+      run: |
+        npm uninstall typescript --no-save
+        npm uninstall tslint --no-save
```
Comment (weswigham) [inline .github/workflows/ci.yml]: I'm _pretty sure_ we can remove this from our scripts, since we don't use `tslint` anymore, XD

**Comment by weswigham** [comment]
BTW @ethomson what's the open source project concurrent job/resource limit look like for actions? (Far as I can tell it's just "free") Part of the reason we use pipelines for a bunch is because we have a lot of extra free extra parallel jobs allotted to us, which helps us do intensive on-demand testing around release periods without getting backlogged.

**Pull request merged**